source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -0,0 +1,732 @@
+=================================
+Connection Monitoring and Pooling
+=================================
+
+:Title: Connection Monitoring and Pooling
+:Author: Dan Aprahamian
+:Advisory Group: Jeff Yemin, Matt Broadstone
+:Approvers: Bernie Hackett, Dan Pasette, Jeff Yemin, Matt Broadstone, Sam Rossi, Scott L'Hommedieu
+:Status: Accepted
+:Type: Standards
+:Minimum Server Version: N/A
+:Last Modified: January 22, 2019
+:Version: 1.0.0
+
+.. contents::
+
+Abstract
+========
+
+Drivers currently support a variety of options that allow users to configure connection pooling behavior. Users are confused by drivers supporting different subsets of these options. Additionally, drivers implement their connection pools differently, making it difficult to design cross-driver pool functionality. By unifying and codifying pooling options and behavior across all drivers, we will increase user comprehension and code base maintainability.
+
+META 
+====
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Definitions
+===========
+
+Endpoint
+~~~~~~~~
+
+For convenience, an Endpoint refers to either a **mongod** or **mongos** instance.
+
+Thread
+~~~~~~
+
+For convenience, a Thread refers to:
+
+-  A shared-address-space process (a.k.a. a thread) in multi-threaded drivers
+-  An Execution Frame / Continuation in asynchronous drivers
+-  A goroutine in Go
+
+Behavioral Description
+======================
+
+Which Drivers this applies to
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This specification is solely concerned with drivers that implement a connection pool. A driver SHOULD implement a connection pool, but is not required to.
+
+Connection Pool Options
+~~~~~~~~~~~~~~~~~~~~~~~
+
+All drivers that implement a connection pool MUST implement and conform to the same MongoClient options. There can be slight deviation in naming to make the options idiomatic to the driver language.
+
+Connection Pool Behaviors
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All driver connection pools MUST provide an API that allows the driver to check out a connection, check in a connection back to the pool, and clear all connections in the pool. This API is for internal use only, and SHOULD NOT be documented as a public API.
+
+Connection Pool Monitoring
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All drivers that implement a connection pool MUST provide an API that allows users to subscribe to events emitted from the pool.
+
+Detailed Design
+===============
+
+.. _connection-pool-options-1:
+
+Connection Pool Options
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Drivers that implement a Connection Pool MUST support the following ConnectionPoolOptions:
+
+.. code:: typescript
+
+    interface ConnectionPoolOptions {
```

Comment:
Can you also update the URI options spec document with this info? I put in TODOs for the links to this spec since since this wasn't merged yet

Reply:
Updated URI spec.